include/openthread/platform/radio.h

```diff
@@ -946,6 +946,36 @@ otError otPlatRadioEnableCsl(otInstance *aInstance, uint32_t aCslPeriod, const o
  */
 void otPlatRadioUpdateCslSampleTime(otInstance *aInstance, uint32_t aCslSampleTime);
 
+/**
+ * Set the max transmit power for a specific channel.
+ *
+ * @param[in]  aInstance    The OpenThread instance structure.
+ * @param[in]  aChannel     The radio channel.
+ * @param[in]  aMaxPower    The max power in dBm.
+ *
+ */
+otError otPlatRadioSetChannelMaxTransmitPower(otInstance *aInstance, uint8_t aChannel, int8_t aMaxPower);
```

Comment:
`aMaxPower == 0x7f` already indicates the channel is not allowed. can we eliminate `otPlatRadioSetChannelSupported()`?

Reply:
I'd like to treat `aMaxPower == 0x7f` as we don't specify the max power and the underlying driver is free to decide the power by self.

Reply:
It seems we should take full control of tx power for certification. Any motivation to give up the tx power limit?

Reply:
When max power table is not passed, we'll use the `otPlatRadioSetTransmitPower` instead for controlling the power. And the max power table will be used for any additional restriction because of regional radio regulations.